app/models/mixins/authentication_mixin.rb

```diff
@@ -158,14 +158,20 @@ def update_authentication(data, options = {})
     # Invoke before callback
     before_update_authentication if self.respond_to?(:before_update_authentication) && options[:save]
 
+    is_amazon = kind_of?(ManageIQ::Providers::Amazon::CloudManager)
+
     data.each_pair do |type, value|
       cred = authentication_type(type)
       current = {:new => nil, :old => nil}
 
       unless value.key?(:userid) && value[:userid].blank?
         current[:new] = {:user => value[:userid], :password => value[:password], :auth_key => value[:auth_key]}
```

Comment:
Could you add the `value[:service_account].presence` here?  If it is nil it shouldn't hurt anything and we can avoid the `is_amazon` check

Reply:
done